src/core/lib/security/credentials/fake/fake_credentials.c

```diff
@@ -49,7 +50,22 @@ static grpc_security_status fake_transport_security_create_security_connector(
     grpc_call_credentials *call_creds, const char *target,
     const grpc_channel_args *args, grpc_channel_security_connector **sc,
     grpc_channel_args **new_args) {
-  *sc = grpc_fake_channel_security_connector_create(call_creds);
+  if (grpc_channel_args_find(args, GRPC_ARG_GRPCLB_BALANCER_NAMES) != NULL) {
+    /* For LB channels, verify the target name passed in matches the
```

Comment:
In the current version of the code, a new security connector is created using the balancer name as the target (see https://github.com/grpc/grpc/pull/9194/files#diff-7817d15cfb7a0b489a3154f5decb3a12R228): the security connector does not perform the lookup, its factory does, feeding the result to the secure connector's constructor. 

THAT BEING SAID, given your suggestion to move the lookup to the connector's check_peer() this test must indeed move to security_connector.c:fake_check_peer (provided check_peer() acquires the ability to read channel args, as mentioned in a previous comment).